frontend/ui/widget/inputtext.lua

```diff
@@ -586,6 +586,12 @@ function InputText:focus()
     self._frame_textwidget.color = Blitbuffer.COLOR_BLACK
 end
 
+local sym_key_map = {
+    ["Q"] = "!", ["W"] = "?", ["E"] = "-", ["R"] = "_", ["T"] = "%", ["Y"] = "=", ["U"] = "7", ["I"] = "8",  ["O"] = "9", ["P"] = "0",
+    ["A"] = "<", ["S"] = ">", ["D"] = "(", ["F"] = ")", ["G"] = "#", ["H"] = "'", ["J"] = "4", ["K"] = "5",  ["L"] = "6",
+    ["Z"] = "{", ["X"] = "}", ["C"] = "[", ["V"] = "]", ["B"] = "1", ["N"] = "2", ["M"] = "3", ["."] = ":", ["AA"] = ";",
+}
```

Comment:
Are these alt symbols shown on the/a kindle keyboard ?
Or did you make them out, and if yes, from what inspiration?
Might be worth mentionning the why and what in a comment.

Reply:
they are not shown as labels (one has to "memorise where they are"). They are 'inspired' by the symbol layer on the virtual keyboard but, I have taken the liberty of making some adjustments since: K3 does not have a numeric row and, I am prioritising the most expected characters for "style tweaks" and text-writing (in English, sorry everybody else, there are just not enough keys)

Reply:
I have a problem, when I press <kbd>Q</kbd> now (for example), this is being printed `!q`, <kbd>A</kbd> prints `<a`, <kbd>Shift</kbd> + <kbd>W</kbd> prints `?W`

Reply:
interesting finding: (let us ignore the whole issue mentioned above for a second)
if I run:

```lua
if Device:hasSymKey() then
    local symkey = sym_key_map[key.key]
    if symkey then
        self:addChars(symkey)
    else
        handled = false
    end
end
```
instead of `if not handled and Device:hasSymKey() then` suddenly something unexpected yet welcome occurs. We no longer need to use the double <kbd> back </kbd>  to "close kb, unfocus" and then "close widget". Suddenly the <kbd>up</kbd>/<kbd>down</kbd> move around the widget, removing focus and making everything the way I described earlier here https://github.com/koreader/koreader/pull/12162#issuecomment-2252571230

more specifically here 

> I have considered the idea of doing away with directly using dpad (when show-vk is unchecked) for text navigation (as we can do it with shift+ dpad ) and using dpad to move focus around dialogue...

is there a way to capture whatever is causing that and wrap it inside a `if hasKB and G_set:false(vk)`?

Reply:
Could i then bother you with the last push here @poire-z ???

Reply:
Well, I don't undestand much of what you're describing. And I had my nightmarish excursion into that last week where nothing made sense.
I don't have the courage and time to go into that again :(
I hope somebody else can accompany you.

Reply:
@poire-z  I solved the issue with this (single alphabet presses would input both a symbol and letter)

```lua
if Device:hasSymKey() then
    local symkey = sym_key_map[key.key]
    if symkey then
        for modifier, flag in pairs(key.modifiers) do
            if modifier == "Sym" and flag then
                self:addChars(symkey)
            end
        end
    else
        handled = false
    end
end
```

it doesn't look pretty but works... do you give it the green light?

Reply:
> do you give it the green light?

_I_ can't. Others may.
To me, all this is hard to grasp, and you'll be adding random stuff that seems to work without really understanding why - and why the simpler most obvious solution doesn't -, making it even more messy and hard to grasp in the future.
(And I don't even understand what "issue" you're talking about - for the same bit of code you're adapting, above it's about back/up/down, now it's about getting 2 chars in the output... Might be me being tired, but not just me :))

Reply:
Ignore the second message https://github.com/koreader/koreader/pull/12162#discussion_r1695332365

issue in ELI5 form: finger presses keyboard key <kbd>Q</kbd>, brain expects the letter `q` to appear on the input text field, instead `!q` appears, that is to say the symbol associated with key[Q] and the letter `q`. same for every letter, so pressing the keys <kbd>H</kbd> <kbd>E</kbd> <kbd>L</kbd> <kbd>L</kbd> <kbd>O</kbd> results in `'h-e6l6l9o` appearing on screen.

> random stuff that seems to work without really understanding why 

I understand why it works, I am only asking if you might have a simpler more elagant approach.

Reply:
> you'll be adding random stuff that seems to work without really understanding why - and why the simpler most obvious solution doesn't -, making it even more messy and hard to grasp in the future.

This.

This is a giant re flag to me, especially in a piece of code nobody really wants to touch already.

Reply:
My advice would be: reduce the scope to stuff you can actually grok and implement without mangling everything in the process ;).